docs/source/essentials/data.md

```diff
@@ -2,22 +2,103 @@
 title: Fetching data with resolvers
 ---
 
-## Overview
-
 GraphQL is the best way to work with data from **any** back-end that your product needs. It is not a mapping of your database, but rather a graph of the data sources and shapes your product is made of. Resolvers are the key to this graph. Each resolver represents a single field, and can be used to fetch data from any source(s) you may have.
 
-## Context
+Resolvers provide the instructions for turning a GraphQL operation into data. Resolvers are organized into a one to one mapping to the fields in a GraphQL schema. This section describes resolvers' organization, every field's default resolver, and their signature.
+
+<h2 id="resolver-map">Resolver map</h2>
+
+In order to respond to queries, a schema needs to have resolve functions for all fields. This collection of functions is called the "resolver map". This map relates the schema fields and types to a function.
+
+```js
+const schema = `
+type Book {
+  title: String
+  author: Author
+}
+
+type Author {
+  books: [Book]
+}
+
+type Query {
+  author: Author
+}
+`;
+
+const resolvers = {
+  Query: {
+    author(root, args, context, info) {
+      return find(authors, { id: args.id });
+    },
+  },
+  Author: {
+    books(author) {
+      return filter(books, { author: author.name });
+    },
+  },
+};
+```
+
+Note that you don't have to put all of your resolvers in one object. Refer to the ["modularizing the schema"](/docs/graphql-tools/generate-schema.html#modularizing) section to learn how to combine multiple resolver maps into one.
+
+<h2 id="type-signature">Resolver type signature</h2>
+
+In addition to the parent resolvers' value, resolvers receive a couple more arguments. The full resolver function signature contains four positional arguments: `(parent, args, context, info)` and can return an object or [Promise](https://codeburst.io/javascript-learn-promises-f1eaa00c5461). Once a promise resolves, then the children resolvers will continue executing. This is useful for fetching data from a [backend]().
```

Comment:
> This is useful for fetching data from a `[backend]()`

I'm not sure you were trying to link anything to the backend here. Were you @peggyrayzis ?

Reply:
No lol thanks for catching this, I deleted the link